src/legacy/core_plugins/console/np_ready/public/interfaces/editor.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { TokenIterator } from './token_iterator';
+
+export interface Position {
+  /**
+   * The line number, not zero-indexed.
+   *
+   * E.g., if given line number 1, this would refer to the first line visible.
+   */
+  lineNumber: number;
+
+  /**
+   * The column number, not zero-indexed.
+   *
+   * E.g., if given column number 1, this would refer to the first character of a column.
+   */
+  column: number;
+}
+
+export interface Range {
+  // TODO: document
+  a: Position;
+  // TODO: document
+  b: Position;
```

Comment:
Would `start` and `end` be clearer? I think I might just be used to those terms from the [Range object](https://developer.mozilla.org/en-US/docs/Web/API/Range) in JavaScript.

Reply:
Good point! I'll update it.